CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFileManager+FileManagement.swift

```diff
@@ -88,29 +88,73 @@ extension CEWorkspaceFileManager {
         )
     }
 
-    /// This function deletes the item or folder from the current project
+    /// This function deletes the item or folder from the current project by moving to Trash
+    /// - Parameters:
+    ///   - file: The file or folder to delete
+    /// - Authors: Paul Ebose
+    public func trash(file: CEWorkspaceFile) {
+        let message: String
+
+        if !file.isFolder || file.isEmptyFolder { // if its a file or an empty folder, call it by its name
+            message = file.name
+        } else {
+            let fileCount = fileManager.enumerator(atPath: file.url.path)?.allObjects.count
+            message = "the \((fileCount ?? 0) + 1) selected items"
+        }
+
+        let moveFileToTrashAlert = NSAlert()
+        moveFileToTrashAlert.messageText = "Do you want to move \(message) to Trash?"
+        moveFileToTrashAlert.informativeText = "This operation cannot be undone."
+        moveFileToTrashAlert.alertStyle = .critical
+        moveFileToTrashAlert.addButton(withTitle: "Move to Trash")
+        moveFileToTrashAlert.buttons.last?.hasDestructiveAction = true
+        moveFileToTrashAlert.addButton(withTitle: "Cancel")
+
+        if moveFileToTrashAlert.runModal() == .alertFirstButtonReturn { // "Move to Trash" button
+            if fileManager.fileExists(atPath: file.url.path) {
+                do {
+                    try fileManager.trashItem(at: file.url, resultingItemURL: nil)
+                } catch {
+                    fatalError(error.localizedDescription)
```

Comment:
Can we just print the error instead of throwing a fatal error? We will eventually move to a proper logging solution.

Reply:
Okay, will do that. I just duplicated the initial delete function and then modified it. Should I also change the delete function's catch block?

Reply:
Don't think it warrants closing the application if a file didn't delete properly. The other ones are fine for now but I think a print is sufficient

Reply:
Okay, I have done that